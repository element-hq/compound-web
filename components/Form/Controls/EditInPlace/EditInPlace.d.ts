import { default as React } from '../../../../../node_modules/react';
import { TextControl } from '../..';
type Props = {
    /**
     * The label for the control
     */
    label: string;
    /**
     * The CSS class name.
     */
    className?: string;
    /**
     * Callback for when the user confirms the change
     */
    onSave?: (e: React.FormEvent<HTMLFormElement>) => Promise<void> | void;
    /**
     * Callback for when the user wishes to cancel the change
     */
    onCancel?: (e: React.FormEvent<HTMLFormElement>) => void;
    /**
     * onInput event handler on the text control
     */
    onInput?: (e: React.ChangeEvent<HTMLInputElement>) => void;
    /**
     * Callback for when the server validation errors should be cleared.
     */
    onClearServerErrors?: () => void;
    /**
     * Whether the field is in an error state according to the server validation.
     *
     * For validation messages, use native validations properties directly, or add custom error messages as children.
     */
    serverInvalid?: boolean;
    /**
     * Label to be displayed by the green check at the bottom. Will only be displayed
     * for 2 seconds after the onSave callback promise resolves successfully.
     */
    savedLabel?: string;
    /**
     * The label for the save button
     */
    saveButtonLabel: string;
    /**
     * The label for the 'in progress' saving caption
     */
    savingLabel: string;
    /**
     * The label for the cancel button
     */
    cancelButtonLabel: string;
    /**
     * Label to be displayed under the input as a help text
     */
    helpLabel?: string;
    /**
     * If true, disabled the entire component to disallow editing.
     */
    disabled?: boolean;
} & React.ComponentProps<typeof TextControl>;
/**
 * A text box with save/cancel buttons that appear when the field is active.
 * Since thios control has its own 'save' button, it should *not* appear as part
 * of a larger form: it exists as its own form that submits separately.
 */
export declare const EditInPlace: React.ForwardRefExoticComponent<Omit<Props, "ref"> & React.RefAttributes<HTMLInputElement>>;
export {};
//# sourceMappingURL=EditInPlace.d.ts.map