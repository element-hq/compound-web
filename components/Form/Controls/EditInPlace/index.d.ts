export { EditInPlace } from './EditInPlace';
//# sourceMappingURL=index.d.ts.map