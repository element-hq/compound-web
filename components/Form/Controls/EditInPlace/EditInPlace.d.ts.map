{"version":3,"file":"EditInPlace.d.ts","sourceRoot":"","sources":["../../../../../src/components/Form/Controls/EditInPlace/EditInPlace.tsx"],"names":[],"mappings":"AAOA,OAAO,KAON,MAAM,OAAO,CAAC;AAOf,OAAO,EAOL,WAAW,EACZ,MAAM,OAAO,CAAC;AAGf,KAAK,KAAK,GAAG;IACX;;OAEG;IACH,KAAK,EAAE,MAAM,CAAC;IAEd;;OAEG;IACH,SAAS,CAAC,EAAE,MAAM,CAAC;IAEnB;;OAEG;IACH,MAAM,CAAC,EAAE,CAAC,CAAC,EAAE,KAAK,CAAC,SAAS,CAAC,eAAe,CAAC,KAAK,OAAO,CAAC,IAAI,CAAC,GAAG,IAAI,CAAC;IAEvE;;OAEG;IACH,QAAQ,CAAC,EAAE,CAAC,CAAC,EAAE,KAAK,CAAC,SAAS,CAAC,eAAe,CAAC,KAAK,IAAI,CAAC;IAEzD;;OAEG;IACH,OAAO,CAAC,EAAE,CAAC,CAAC,EAAE,KAAK,CAAC,WAAW,CAAC,gBAAgB,CAAC,KAAK,IAAI,CAAC;IAE3D;;OAEG;IACH,mBAAmB,CAAC,EAAE,MAAM,IAAI,CAAC;IAEjC;;;;OAIG;IACH,aAAa,CAAC,EAAE,OAAO,CAAC;IAExB;;;OAGG;IACH,UAAU,CAAC,EAAE,MAAM,CAAC;IAEpB;;OAEG;IACH,eAAe,EAAE,MAAM,CAAC;IAExB;;OAEG;IACH,WAAW,EAAE,MAAM,CAAC;IAEpB;;OAEG;IACH,iBAAiB,EAAE,MAAM,CAAC;IAE1B;;OAEG;IACH,SAAS,CAAC,EAAE,MAAM,CAAC;IAEnB;;OAEG;IACH,QAAQ,CAAC,EAAE,OAAO,CAAC;CACpB,GAAG,KAAK,CAAC,cAAc,CAAC,OAAO,WAAW,CAAC,CAAC;AAyD7C;;;;GAIG;AACH,eAAO,MAAM,WAAW,6FAyMvB,CAAC"}