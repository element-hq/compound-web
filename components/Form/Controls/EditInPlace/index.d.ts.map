{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../src/components/Form/Controls/EditInPlace/index.ts"],"names":[],"mappings":"AAOA,OAAO,EAAE,WAAW,EAAE,MAAM,eAAe,CAAC"}