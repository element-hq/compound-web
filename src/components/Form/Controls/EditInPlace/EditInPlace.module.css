/*
Copyright 2024 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.container {
  min-block-size: 110px;
}

.label {
  font-size: 15px;
  font-weight: 500;
  margin-block-end: 4px;
}

.controls {
  display: flex;
  gap: 15px;
}

.button-group {
  display: none;
  position: relative;
  inset-block-start: 5px;
  grid-template-columns: 1fr 1fr;
  gap: 7px;
}

.button {
  display: flex;
  inline-size: 36px;
  block-size: 36px;
  border: 1px solid var(--cpd-color-border-interactive-secondary);
  border-radius: 20px;
  background-color: transparent;
  text-align: center;
  cursor: pointer;
}

.primary-button {
  background-color: var(--cpd-color-bg-action-primary-rest);
  border-color: var(--cpd-color-bg-action-primary-rest);

  svg {
    color: var(--cpd-color-icon-on-solid-primary);
  }
}

.primary-button-disabled {
  background-color: var(--cpd-color-bg-subtle-primary);
  border-color: var(--cpd-color-bg-subtle-primary);

  svg {
    color: var(--cpd-color-icon-disabled);
  }
}

.button svg {
  inline-size: 24px;
  block-size: 24px;
  margin: auto;
}

.control {
  inline-size: 100%;
}

.container:focus-within .button-group {
  display: inline-grid;
}

.container-error .control {
  border-color: var(--cpd-color-border-critical-primary);
}

.container-error .label {
  color: var(--cpd-color-text-critical-primary);
}

.caption-line {
  margin-block-start: var(--cpd-space-2x);
}

.caption-icon {
  display: inline-block;
  vertical-align: middle;
  inline-size: 20px;
  block-size: 20px;
  margin-inline-end: var(--cpd-space-2x);
}

.caption-icon-error {
  color: var(--cpd-color-icon-critical-primary);
}

.caption-icon-saved {
  background-color: var(--cpd-color-icon-success-primary);
  border-radius: 20px;
  text-align: center;
  margin-inline-end: var(--cpd-space-2x);

  svg {
    color: var(--cpd-color-icon-on-solid-primary);
  }
}

.caption-text {
  vertical-align: middle;
  font-size: 13px;
}

.caption-text-error {
  color: var(--cpd-color-text-critical-primary);
}

.caption-text-saved {
  color: var(--cpd-color-text-success-primary);
}
