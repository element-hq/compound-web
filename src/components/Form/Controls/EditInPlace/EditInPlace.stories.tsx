/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import React from "react";

import { EditInPlace } from "./";
import { type Meta, type StoryObj } from "@storybook/react-vite";
import { expect, userEvent, within } from "storybook/test";
import { ErrorMessage } from "../../Message";
import { TooltipProvider } from "../../../Tooltip/TooltipProvider";

type Props = { invalid?: boolean } & React.ComponentProps<typeof EditInPlace>;

const meta = {
  title: "Form/Controls/EditInPlace",
  component: EditInPlace,
  tags: ["autodocs"],
  parameters: {
    controls: {
      include: [
        "onChange",
        "onSave",
        "onCancel",
        "onClearServerErrors",
        "defaultValue",
        "error",
        "serverInvalid",
        "savedLabel",
        "saveButtonLabel",
        "cancelButtonLabel",
        "helpLabel",
        "disabled",
      ],
    },
    design: {
      type: "figma",
      url: "https://www.figma.com/file/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?type=design&node-id=4335-2016&mode=design&t=BvxRca0YDRaq20IR-0",
    },
  },
  argTypes: {
    label: {
      type: "string",
    },
    defaultValue: {
      type: "string",
    },
    onChange: {
      action: "changed",
    },
    onCancel: {
      action: "cancelled",
    },
    onClearServerErrors: {
      action: "cleared server errors",
    },
    serverInvalid: {
      type: "boolean",
    },
    helpLabel: {
      type: "string",
    },
    savedLabel: {
      type: "string",
    },
    saveButtonLabel: {
      type: "string",
    },
    savingLabel: {
      type: "string",
    },
    cancelButtonLabel: {
      type: "string",
    },
    disabled: {
      type: "boolean",
    },
  },
  render: ({ ...restArgs }) => (
    <TooltipProvider>
      <EditInPlace {...restArgs} />
    </TooltipProvider>
  ),
  args: {
    label: "Label",
    onSave: () => new Promise((resolve) => setTimeout(resolve, 1000)),
    savedLabel: "Saved",
    saveButtonLabel: "Save",
    cancelButtonLabel: "Cancel",
    savingLabel: "Saving…",
  },
} satisfies Meta<Props>;
export default meta;

type Story = StoryObj<typeof meta>;

export const Empty: Story = {};

export const WithText: Story = {
  args: {
    defaultValue: "Hello, Computer",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const input = canvas.getByRole("textbox");
    await userEvent.clear(input);
    await userEvent.type(input, "Hello, Computer");
  },
};

export const Saving: Story = {
  tags: ["axe-exclude"],
  args: {
    defaultValue: "Hello",
    onSave: () => new Promise(() => {}),
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const input = canvas.getByRole("textbox");
    await userEvent.clear(input);
    await userEvent.type(input, "Hello");
    const save = canvas.getByRole("button", { name: "Save" });
    await userEvent.click(save);
    await expect(canvas.getByText("Saving…")).toBeInTheDocument();
  },
};

export const WithError: Story = {
  args: {
    serverInvalid: true,
    children: <ErrorMessage>I am a teapot</ErrorMessage>,
  },
};

export const WithHelpText: Story = {
  args: {
    helpLabel: "Optional help text.",
  },
};
