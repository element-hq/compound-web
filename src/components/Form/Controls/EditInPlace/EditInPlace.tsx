/*
Copyright 2024 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import classnames from "classnames";
import React, { FormEvent, forwardRef, useCallback, useRef } from "react";
import styles from "./EditInPlace.module.css";
import { TextInput } from "../Text";
import useId from "../../../../utils/useId";

import CheckIcon from "@vector-im/compound-design-tokens/icons/check.svg";
import CancelIcon from "@vector-im/compound-design-tokens/icons/close.svg";
import ErrorIcon from "@vector-im/compound-design-tokens/icons/error.svg";

type Props = {
  /**
   * The label for the control
   */
  label: string;

  /**
   * The CSS class name.
   */
  className?: string;

  /**
   * The content of the text box
   */
  value: string;

  /**
   * Callback for when the user confirms the change
   */
  onSave: () => Promise<void>;

  /**
   * Calback for when the user wishes to cancel the change
   */
  onCancel: () => void;

  /**
   * Error message to be displayed below the box. If supplied, will disable the
   * save button.
   */
  error?: string;

  /**
   * Label to be displayed by the green check at the bottom. Will only be displayed
   * for 2 seconds after the onSave callback promise resolves successfully.
   */
  savedLabel?: string;

  /**
   * The label for the save button
   */
  saveButtonLabel: string;

  /**
   * True to disable the save button, false to enasble.
   * Default: false (enabled)
   */
  disableSaveButton?: boolean;

  /**
   * The label for the cancel button
   */
  cancelButtonLabel?: string;
} & React.ComponentProps<typeof TextInput>;

/**
 * A text box with save/cancel buttons that appear when the field is active.
 * Since thios control has its own 'save' button, it should *not* appear as part
 * of a larger form: it exists as its own form that submits separately.
 */
export const EditInPlace = forwardRef<HTMLInputElement, Props>(
  function EditInPlace(
    {
      className,
      label,
      onSave,
      onCancel,
      saveButtonLabel,
      cancelButtonLabel,
      disableSaveButton,
      error,
      savedLabel,
      ...props
    },
    ref,
  ) {
    const id = useId();
    const labelId = useId();
    const errorTextId = useId();
    const classes = classnames(styles.container, className, {
      [styles["container-error"]]: Boolean(error),
    });

    const [showSaved, setShowSaved] = React.useState(false);

    const hideTimer = useRef<NodeJS.Timeout | null>(null);

    const saveButtonRef = useRef<HTMLButtonElement | null>(null);
    const cancelButtonRef = useRef<HTMLButtonElement | null>(null);

    React.useEffect(() => {
      return () => {
        if (hideTimer.current) clearTimeout(hideTimer.current);
      };
    }, []);

    const onFormSubmit = useCallback(
      async (e: FormEvent) => {
        e.preventDefault();
        try {
          await onSave();
          saveButtonRef.current?.blur();
          setShowSaved(true);
          hideTimer.current = setTimeout(() => {
            setShowSaved(false);
          }, 2000);
        } catch (e) {
          // We don't really need to do anything here, we just don't want to display the
          // 'saved' label, obviously. The user of the component can update the error to
          // show what failed.
        }
      },
      [setShowSaved, onSave, hideTimer],
    );

    const onCancelButtonClicked = useCallback(() => {
      cancelButtonRef.current?.blur();
      onCancel();
    }, [cancelButtonRef, onCancel]);

    return (
      <form className={classes} id={id} onSubmit={onFormSubmit}>
        <div className={styles.label} id={labelId}>
          {label}
        </div>
        <div className={styles.controls}>
          <TextInput
            ref={ref}
            {...props}
            className={styles.control}
            aria-labelledby={labelId}
            aria-invalid={Boolean(error)}
            aria-errormessage={error ? errorTextId : undefined}
          />
          <div className={styles["button-group"]}>
            <button
              type="submit"
              className={classnames(styles.button, styles["primary-button"], {
                [styles["primary-button-disabled"]]: disableSaveButton,
              })}
              ref={saveButtonRef}
              aria-controls={id}
              aria-label={saveButtonLabel}
              disabled={disableSaveButton}
            >
              <CheckIcon />
            </button>
            <button
              type="button"
              role="button"
              ref={cancelButtonRef}
              className={styles.button}
              onClick={onCancelButtonClicked}
              aria-controls={id}
              aria-label={cancelButtonLabel}
            >
              <CancelIcon />
            </button>
          </div>
        </div>
        {error && (
          <div className={styles["caption-line"]}>
            <ErrorIcon
              className={classnames(
                styles["caption-icon"],
                styles["caption-icon-error"],
              )}
            />
            <span
              id={errorTextId}
              className={classnames(
                styles["caption-text"],
                styles["caption-text-error"],
              )}
            >
              {error}
            </span>
          </div>
        )}
        {savedLabel && showSaved && (
          <div className={styles["caption-line"]}>
            <div
              className={classnames(
                styles["caption-icon"],
                styles["caption-icon-saved"],
              )}
            >
              <CheckIcon />
            </div>
            <span
              className={classnames(
                styles["caption-text"],
                styles["caption-text-saved"],
              )}
            >
              {savedLabel}
            </span>
          </div>
        )}
      </form>
    );
  },
);
