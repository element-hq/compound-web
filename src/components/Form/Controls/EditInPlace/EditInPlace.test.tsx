/*
Copyright 2024 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import { describe, it, expect, vi, afterEach } from "vitest";
import React, { ChangeEvent } from "react";
import { render, screen } from "@testing-library/react";

import { EditInPlace } from "./EditInPlace";
import { Root, Field, Control } from "@radix-ui/react-form";
import { userEvent } from "@storybook/test";
import { act } from "react-dom/test-utils";

type EditInPlaceTestProps = {
  error?: string;
  value?: string;
  disableSaveButton?: boolean;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  onSave?: () => Promise<void>;
  onCancel?: () => void;
};

describe("EditInPlace", () => {
  const EditInPlaceTest = (props: EditInPlaceTestProps) => (
    <Root>
      <Field name="editInPlace">
        <Control asChild>
          <EditInPlace
            label="Edit Me"
            value={props.value ?? "Edit this text"}
            error={props.error}
            onChange={props.onChange ?? (() => {})}
            onSave={props.onSave ?? (() => Promise.resolve())}
            onCancel={props.onCancel ?? (() => {})}
            saveButtonLabel="Save"
            cancelButtonLabel="Cancel"
            savedLabel={"Saved"}
            disableSaveButton={props.disableSaveButton}
          />
        </Control>
      </Field>
    </Root>
  );

  afterEach(() => {
    vi.useRealTimers();
  });

  it("renders", () => {
    const { asFragment } = render(<EditInPlaceTest />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onChange when text edited", async () => {
    let value;
    const onChange = vi
      .fn()
      .mockImplementation((e: ChangeEvent<HTMLInputElement>) => {
        value = e.target.value;
      });
    render(<EditInPlaceTest onChange={onChange} />);

    await act(async () => {
      const input = screen.getByRole("textbox");
      // nb. we don't test updating the value here so we only type one character
      await userEvent.type(input, "!");
    });

    expect(onChange).toHaveBeenCalled();
    expect(value).toEqual("Edit this text!");
  });

  it("field is valid if no error specified", () => {
    render(<EditInPlaceTest />);

    const input = screen.getByRole("textbox");
    expect(input).toBeValid();
  });

  it("renders error icon and text if error provided", () => {
    render(<EditInPlaceTest error="Missing Left Falangey" />);

    const input = screen.getByRole("textbox");
    expect(input).toBeInvalid();
    expect(input).toHaveAttribute("aria-errormessage");

    const errorText = screen.getByText("Missing Left Falangey");
    expect(errorText).toBeInTheDocument();

    expect(errorText.id).toEqual(input.getAttribute("aria-errormessage"));
  });

  it("should disable save button if told to", () => {
    render(<EditInPlaceTest disableSaveButton={true} />);

    const saveButton = screen.getByRole("button", { name: "Save" });
    expect(saveButton).toBeDisabled();
  });

  it("enables save button by default", () => {
    render(<EditInPlaceTest value="Something else" />);

    expect(screen.getByRole("button", { name: "Save" })).toBeEnabled();
  });

  it("calls save callback on save button click", async () => {
    const onSave = vi.fn();
    render(<EditInPlaceTest onSave={onSave} value="Changed" />);

    await act(async () => {
      await userEvent.click(screen.getByRole("button", { name: "Save" }));
    });

    expect(onSave).toHaveBeenCalled();
  });

  it("calls save callback if enter pressed in the text box", async () => {
    const onSave = vi.fn();
    render(<EditInPlaceTest onSave={onSave} value="Changed" />);

    await act(async () => {
      await userEvent.type(screen.getByRole("textbox"), "{enter}");
    });

    expect(onSave).toHaveBeenCalled();
  });

  it("calls onCancel when cancel button pressed", async () => {
    const onCancel = vi.fn();
    render(<EditInPlaceTest onCancel={onCancel} value="Changed" />);

    await userEvent.click(screen.getByRole("button", { name: "Cancel" }));

    expect(onCancel).toHaveBeenCalled();
  });

  it("unfocuses the input when cancel button pressed", async () => {
    render(<EditInPlaceTest value="Changed" />);

    const input = screen.getByRole("textbox");
    await userEvent.click(screen.getByRole("button", { name: "Cancel" }));

    expect(document.activeElement).not.toEqual(input);
  });

  it("unfocuses the input when the save calllback promise resolves", async () => {
    render(<EditInPlaceTest value="Changed" />);

    await act(async () => {
      await userEvent.click(screen.getByRole("button", { name: "Save" }));
    });

    const input = screen.getByRole("textbox");
    expect(document.activeElement).not.toEqual(input);
  });

  it("displays saved label for 2 seconds after save", async () => {
    vi.useFakeTimers();
    const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime });

    render(<EditInPlaceTest value="Changed" />);

    expect(screen.queryByText("Saved")).not.toBeInTheDocument();

    await act(async () => {
      await user.click(screen.getByRole("button", { name: "Save" }));
    });

    expect(screen.getByText("Saved")).toBeInTheDocument();

    act(() => {
      vi.advanceTimersByTime(1900);
    });

    expect(screen.queryByText("Saved")).toBeInTheDocument();

    act(() => {
      vi.advanceTimersByTime(200);
    });

    expect(screen.queryByText("Saved")).not.toBeInTheDocument();
  });

  it("does not call onSave if cancel pressed", async () => {
    const onSave = vi.fn();
    render(<EditInPlaceTest onSave={onSave} value="Changed" />);

    await userEvent.click(screen.getByRole("button", { name: "Cancel" }));

    expect(onSave).not.toHaveBeenCalled();
  });
});
