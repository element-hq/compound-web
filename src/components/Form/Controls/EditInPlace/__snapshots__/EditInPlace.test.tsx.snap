// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditInPlace > renders 1`] = `
<DocumentFragment>
  <form>
    <div>
      <div
        class="_label_980156"
        id=":r2:"
      >
        Edit Me
      </div>
      <div
        class="_controls_980156"
      >
        <input
          aria-invalid="false"
          aria-labelledby=":r2:"
          class="_control_a839aa _control_980156"
          id="radix-:r0:"
          name="editInPlace"
          title=""
          value="Edit this text"
        />
        <div
          class="_button-group_980156"
        >
          <button
            aria-controls=":r1:"
            aria-label="Save"
            class="_button_980156 _primary-button_980156"
            type="submit"
          >
            <svg
              class="cpd-icon"
              fill="currentColor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.55 17.575c-.133 0-.258-.02-.375-.063a.878.878 0 0 1-.325-.212L4.55 13c-.183-.183-.27-.42-.263-.713.009-.291.105-.529.288-.712a.948.948 0 0 1 .7-.275.95.95 0 0 1 .7.275L9.55 15.15l8.475-8.475c.183-.183.42-.275.712-.275s.53.092.713.275c.183.183.275.42.275.712s-.092.53-.275.713l-9.2 9.2c-.1.1-.208.17-.325.212a1.106 1.106 0 0 1-.375.063Z"
              />
            </svg>
          </button>
          <button
            aria-controls=":r1:"
            aria-label="Cancel"
            class="_button_980156"
            role="button"
            type="button"
          >
            <svg
              class="cpd-icon"
              fill="currentColor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.293 6.293a1 1 0 0 1 1.414 0L12 10.586l4.293-4.293a1 1 0 1 1 1.414 1.414L13.414 12l4.293 4.293a1 1 0 0 1-1.414 1.414L12 13.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L10.586 12 6.293 7.707a1 1 0 0 1 0-1.414Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
